`no-unknown-classes`: `strings` matcher flags string literals passed to nested function calls inside `clsx`/`cn`

### Description

## Description

`better-tailwindcss/no-unknown-classes` reports string literals that are arguments to a **nested function call** inside a callee (e.g. `clsx`), even though those strings are not class names.

The default matcher for `clsx`/`cn` is `{ match: "strings" }`. Per the [advanced configuration docs](https://github.com/schoero/eslint-plugin-better-tailwindcss/blob/main/docs/configuration/advanced.md), the `strings` matcher should collect **only direct string literals** — strings inside nested structures are supposed to require explicit matchers (`objectValues`, `anonymousFunctionReturn`, …). But a string passed to a nested `CallExpression` (which is neither a direct argument nor an object value / anonymous-function return) is still collected and linted.

This appears to be the same class of over-matching as the previously fixed #226 and #210 (under the rule's former name `no-unregistered-classes`), but for nested call arguments specifically. I couldn't find an existing open issue for this case.

### Flavor

TSX

### Code Input

```tsx
import clsx from "clsx";

// `helper`'s argument is a mode string, NOT a class name
const helper = (mode: string) => (mode === "a" ? "flex" : "block");

const Component = () => (
  <div className={clsx(helper("not-a-classname"), "flex")}>x</div>
);
```

### Expected Behavior

No error. "not-a-classname" is an argument to `helper`, not a direct string literal argument of clsx(), so the `strings` matcher should ignore it.
(Only "flex" should be considered a class.)

### Actual Behavior

```tsx
"not-a-classname" is reported as an unknown class:


//   Unknown class detected: not-a-classname  better-tailwindcss/no-unknown-classes
```

### Reproduction URL

_No response_

### ESLint Log

```shell
/repro/Component.tsx
  6:24  error  Unknown class detected: not-a-classname  better-tailwindcss/no-unknown-classes

✖ 1 problem (1 error, 0 warnings)
```

### ESLint Config

```typescript
import betterTailwind from "eslint-plugin-better-tailwindcss";

export default [
  {
    files: ["**/*.{ts,tsx}"],
    extends: [betterTailwind.configs.correctness],
    settings: {
      "better-tailwindcss": {
        entryPoint: "src/styles/globals.css",
      },
    },
  },
];
```

### Versions

- ESLint: 9.39.4
- Parser: typescript-eslint (flat config)
- Plugin: eslint-plugin-better-tailwindcss 4.6.0
- Node: 24.12.0
- Tailwind CSS: 4.3.1

--- a/src/utils/matchers.test.ts
+++ b/src/utils/matchers.test.ts
@@ -299,6 +299,22 @@ describe("matchers", () => {
       });
     });
 
+    it("should not lint strings passed to nested function calls", () => {
+      lint(noUnnecessaryWhitespace, {
+        invalid: [
+          {
+            jsx: `defined(helper("  ignore  "), " lint ");`,
+            jsxOutput: `defined(helper("  ignore  "), "lint");`,
+
+            errors: 2,
+            options: [{
+              callees: [["defined", [{ match: MatcherType.String }]]]
+            }]
+          }
+        ]
+      });
+    });
+
   });
 
   describe("variables", () => {

--- a/src/utils/matchers.test.ts
+++ b/src/utils/matchers.test.ts
@@ -305,6 +305,10 @@ describe("matchers", () => {
           {
             jsx: `defined(helper("  ignore  "), " lint ");`,
             jsxOutput: `defined(helper("  ignore  "), "lint");`,
+            svelte: `<script>defined(helper("  ignore  "), " lint ");</script>`,
+            svelteOutput: `<script>defined(helper("  ignore  "), "lint");</script>`,
+            vue: `<script>defined(helper("  ignore  "), " lint ");</script>`,
+            vueOutput: `<script>defined(helper("  ignore  "), "lint");</script>`,
 
             errors: 2,
             options: [{

--- a/src/utils/matchers.ts
+++ b/src/utils/matchers.ts
@@ -18,8 +18,6 @@ import type { GenericNodeWithParent } from "better-tailwindcss:utils/utils.js";
 export function getLiteralNodesByMatchers<Node>(ctx: Rule.RuleContext, node: unknown, matcherFunctions: MatcherFunctions): Node[] {
   if(!isGenericNodeWithParent(node)){ return []; }
 
-  const nestedLiterals = findMatchingNestedNodes<Node>(node, matcherFunctions);
-
   const self = matcherFunctions.reduce<Node[]>((self, matcherFunction) => {
     const result = matcherFunction(node);
 
@@ -27,18 +25,37 @@ export function getLiteralNodesByMatchers<Node>(ctx: Rule.RuleContext, node: unk
       return self;
     } else if(result === MATCHER_RESULT.MATCH){
       return [node as Node, ...self];
-    } else if(result === MATCHER_RESULT.UNCROSSABLE_BOUNDARY){
-      return self;
-    } else if(Array.isArray(result)){
-      self.push(...findMatchingNestedNodes<Node>(node, result));
     }
+
     return self;
   }, []);
+
+  const nestedMatcherFunctions = matcherFunctions.reduce<MatcherFunctions>((nestedMatcherFunctions, matcherFunction) => {
+    const result = matcherFunction(node);
+
+    if(result === MATCHER_RESULT.NO_MATCH || result === MATCHER_RESULT.MATCH){
+      nestedMatcherFunctions.push(matcherFunction);
+      return nestedMatcherFunctions;
+    }
+
+    if(result === MATCHER_RESULT.UNCROSSABLE_BOUNDARY){
+      return nestedMatcherFunctions;
+    }
+
+    nestedMatcherFunctions.push(...result);
+    return nestedMatcherFunctions;
+  }, []);
+
+  const nestedLiterals = findMatchingNestedNodes<Node>(node, nestedMatcherFunctions);
 
   return [...nestedLiterals, ...self];
 }
 
 function findMatchingNestedNodes<Node>(node: GenericNodeWithParent, matcherFunctions: MatcherFunctions) {
+  if(matcherFunctions.length === 0){
+    return [];
+  }
+
   return Object.entries(node).reduce<Node[]>((matchedNodes, [key, value]) => {
     if(!value || typeof value !== "object" || key === "parent"){
       return matchedNodes;

--- a/src/utils/matchers.ts
+++ b/src/utils/matchers.ts
@@ -18,33 +18,33 @@ import type { GenericNodeWithParent } from "better-tailwindcss:utils/utils.js";
 export function getLiteralNodesByMatchers<Node>(ctx: Rule.RuleContext, node: unknown, matcherFunctions: MatcherFunctions): Node[] {
   if(!isGenericNodeWithParent(node)){ return []; }
 
-  const self = matcherFunctions.reduce<Node[]>((self, matcherFunction) => {
+  const { nestedMatcherFunctions, self } = matcherFunctions.reduce<{
+    nestedMatcherFunctions: MatcherFunctions;
+    self: Node[];
+  }>((matchedNodes, matcherFunction) => {
     const result = matcherFunction(node);
 
     if(result === MATCHER_RESULT.NO_MATCH){
-      return self;
-    } else if(result === MATCHER_RESULT.MATCH){
-      return [node as Node, ...self];
+      matchedNodes.nestedMatcherFunctions.push(matcherFunction);
+      return matchedNodes;
     }
 
-    return self;
-  }, []);
-
-  const nestedMatcherFunctions = matcherFunctions.reduce<MatcherFunctions>((nestedMatcherFunctions, matcherFunction) => {
-    const result = matcherFunction(node);
-
-    if(result === MATCHER_RESULT.NO_MATCH || result === MATCHER_RESULT.MATCH){
-      nestedMatcherFunctions.push(matcherFunction);
-      return nestedMatcherFunctions;
+    if(result === MATCHER_RESULT.MATCH){
+      matchedNodes.self.unshift(node as Node);
+      matchedNodes.nestedMatcherFunctions.push(matcherFunction);
+      return matchedNodes;
     }
 
     if(result === MATCHER_RESULT.UNCROSSABLE_BOUNDARY){
-      return nestedMatcherFunctions;
+      return matchedNodes;
     }
 
-    nestedMatcherFunctions.push(...result);
-    return nestedMatcherFunctions;
-  }, []);
+    matchedNodes.nestedMatcherFunctions.push(...result);
+    return matchedNodes;
+  }, {
+    nestedMatcherFunctions: [],
+    self: []
+  });
 
   const nestedLiterals = findMatchingNestedNodes<Node>(node, nestedMatcherFunctions);
 

